src/test/java/org/broadinstitute/hellbender/utils/help/TestWDLTool.java

```diff
@@ -0,0 +1,187 @@
+
+    @PositionalArguments(doc = "Positional args doc")
+    @WorkflowInput(requiredCompanions={"posDictionary", "posIndex"})
+    public List<File> positionalListFileInput;
+
+    // required Files
+    @Argument(fullName = "requiredScalarFileInput",
+            shortName = "requiredScalarFileInput",
+            doc = "requiredScalarFileInput doc",
+            optional = false)
+    @WorkflowInput(requiredCompanions={"requiredScalarFileInputDictionary", "requiredScalarFileInputIndex"}, localizationOptional = true)
+    public File requiredScalarFileInput;
+
+    @Argument(fullName = "requiredListFileInput",
+            shortName = "requiredListFileInput",
+            doc = "requiredListFileInput doc",
+            optional = false)
+    @WorkflowInput(requiredCompanions={"requiredListFileInputDictionary", "requiredListFileInputIndex"}, localizationOptional = true)
+    public List<File> requiredListFileInput;
+
+    @Argument(fullName = "requiredScalarFileOutput",
+            shortName = "requiredScalarFileOutput",
+            doc = "requiredScalarFileOutput doc",
+            optional = false)
+    @WorkflowOutput(requiredCompanions={"requiredScalarFileOutputDictionary", "requiredScalarFileOutputIndex"})
+    public File requiredScalarFileOutput;
+
+    @Argument(fullName = "requiredListFileOutput",
+            shortName = "requiredListFileOutput",
+            doc = "requiredListFileOutput doc",
+            optional = false)
+    @WorkflowOutput(requiredCompanions={"requiredListFileOutputDictionary", "requiredListFileOutputIndex"})
+    public List<File> requiredListFileOutput;
+
+
+    // optional Files
+    @Argument(fullName = "optionalScalarFileInput",
+            shortName = "optionalScalarFileInput",
+            doc = "optionalScalarFileInput doc",
+            optional = true)
+    @WorkflowInput(requiredCompanions={"optionalScalarFileInputDictionary", "optionalScalarFileInputIndex"})
+    public File optionalScalarFileInput;
+
+    @Argument(fullName = "optionalListFileInput",
+            shortName = "optionalListFileInput",
+            doc = "optionalListFileInput doc",
+            optional = true)
+    @WorkflowInput(requiredCompanions={"optionalListFileInputDictionary", "optionalListFileInputIndex"})
+    public List<File> optionalListFileInput;
+
+    @Argument(fullName = "optionaldScalarFileOutput",
+            shortName = "optionalScalarFileOutput",
+            doc = "optionalScalarFileOutput doc",
+            optional = true)
+    @WorkflowOutput(requiredCompanions={"optionalScalarFileOutputDictionary", "optionalScalarFileOutputIndex"})
+    public File optionalScalarFileOutput;
+
+    @Argument(fullName = "optionaldListFileOutput",
+            shortName = "optionalListFileOutput",
+            doc = "optionalListFileOutput doc",
+            optional = true)
+    @WorkflowOutput(requiredCompanions={"optionalListFileOutputDictionary", "optionalListFileOutputIndex"})
```

Comment:
No test coverage for `optionalCompanions`?

Reply:
Oh wow - done. I also added some mixed companions cases.